xarray/plot/plot.py

```diff
@@ -221,18 +222,38 @@ def _infer_line_data(darray, x, y, hue):
 
         if y is None:
             xname, huename = _infer_xy_labels(darray=darray, x=x, y=hue)
-            yname = darray.name
-            xplt = darray.coords[xname]
-            yplt = darray.transpose(xname, huename)
+            xplt = darray[xname]
+            if xplt.ndim > 1:
+                if huename in darray.dims:
+                    otherdim = darray.dims[
+                        np.mod(darray.dims.index(huename) + 1, 2)]
+                    yplt = darray.transpose(otherdim, huename)
+                    xplt = xplt.transpose(otherdim, huename)
+                else:
+                    raise ValueError('If x or y are 2D, hue must be a dimension' +
```

Comment:
-> ``hue must be a dimension name '`` (mind the space character at the end of lines, also line  below)